### PR TITLE
[Mobile] Fix gallery style issues

### PR DESCRIPTION
**Pull request by mkevins**
Related gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1766

## Description
This PR fixes some known style issues listed here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1674#issue-536788672. It adjusts the appender color in the gallery block, and fixes a fence-post issue within gallery settings BottomSheet.

## How has this been tested?
* Select a gallery block with images
* Make sure the appender (+) color matches the settings (gear) icon color
* Tap the gear icon
* Last cell in gallery settings bottom sheet shouldn’t have a separator (easier to see on iOS)

## Screenshots 

Example of appender color mismatch:

image

Example of separator problem:

image
</details>

## Types of changes
A mobile specific prop constant is introduced to specify separator type = 'none' for the last item in the `InspectorControls`. Since the last item can vary depending on whether Image size options are displayed, the control preceding "Images Size" control (i.e. "Link To") will conditionally have its separator type set to 'none'. Ideally, this fence-post logic would not be necessary here, and could be handled somewhere else, such as the controls' container, however, this may require more investigation.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/block-editor/src/components/media-placeholder/styles.native.scss [hunk 1]
```diff
@@ -60,13 +60,13 @@
 
 .addMediaButton {
 	color: $white;
-	background-color: $dark-gray-500;
+	background-color: #7b9ab1;
```
Comment (koke) [inline packages/block-editor/src/components/media-placeholder/styles.native.scss]: I see this hex code is present in `_colors.scss` as `$toolbar-button`, we should replace it with the variable in all the places it's being used.

File: packages/block-library/src/gallery/edit.js [hunk 2]
```diff
@@ -340,6 +344,11 @@ class GalleryEdit extends Component {
 		}
 
 		const imageSizeOptions = this.getImagesSizeOptions();
+		const shouldShowSizeOptions = hasImages && ! isEmpty( imageSizeOptions );
+		// This is needed to fix a separator fence-post issue on mobile.
+		const mobileLinkToProps = shouldShowSizeOptions ?
+			MOBILE_CONTROL_PROPS :
+			MOBILE_CONTROL_PROPS_SEPARATOR_NONE;
```
Comment (koke) [inline packages/block-library/src/gallery/edit.js]: This isn't very nice, but I agree that we need a better general solution for the separators. For this one it's ok

**Review by koke — APPROVED** [review]
Other than the SCSS variable, this is looking good 👌

File: packages/block-editor/src/components/media-placeholder/styles.native.scss (hunk 1 above)
Comment (mkevins) [reply packages/block-editor/src/components/media-placeholder/styles.native.scss]: Nice find! I didn't realize we already had this as a variable. I've updated to use the `$toolbar-button` variable (except in `packages/components/src/button/index.native.js` which uses `StyleSheet.create`, where I left a comment instead).

I was tempted to change a few other hard-coded color values to their respective variable names, but quickly realized there are a lot, and such a large change is probably deserving of its own PR. I can open a tech debt issue if one does not already exist.

**Comment by mkevins** [comment]
Thanks for reviewing @koke :+1:

File: packages/block-editor/src/components/media-placeholder/styles.native.scss (hunk 1 above)
Comment (mkevins) [reply packages/block-editor/src/components/media-placeholder/styles.native.scss]: I opened an issue to track this here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1783

**Pull request merged**